packages/plugins/shared/src/config.ts

```diff
@@ -19,12 +19,22 @@ const isYarnBuildConfiguration = t.isObject({
   ignoreFile: t.isString(),
 });
 
+const isYarnBuildManifestConfiguration = t.isObject({
+  input: t.isOptional(t.isOneOf([t.isString(), t.isArray(t.isString())])),
+  output: t.isOptional(t.isOneOf([t.isString(), t.isArray(t.isString())])),
```

Comment:
OK, sounds good to me. Let my try to summarize the impacts for the current implementation:
1. no explicit output config in `package.json`
   - use default config, which is null -> does nothing here
   - use package.json fields
   - use tsconfig.json or custom tsconfig fields
   - none of these provided/found -> no output checking --> rebuild only if input changes
2. explicit output config in `package.json` with file, pattern, etc.
   - use provided output pattern
   - tsconfig configured explicitly -> use this, too (see my commment below)
3. explicit output config in `package.json` with value `null`
   - tsconfig configured explicitly --> use this (see my commment below)
   - otherwise no output checking -> rebuild only if input changes

To be honest I'm not sure if the combination of explicit input/output and tsconfig is a good idea. Setting a custom tsconfig is necessary for me, but we shouldn't merge input and output config with tsconfig values, if the inputs/outputs are configured explicitly. Defining input/output should be some kind of last resort to control the behaviour of yarn.build. That's why I think that we shouldn't enhance/change the input/output config, when it's defined expliclity in package.json.